src/lib/loaders/getCanonicalResourceDirectiveField.ts

```diff
@@ -2,31 +2,23 @@ import { FieldNode, visit, BREAK } from "graphql"
 
 export const getCanonicalResourceDirectiveForField = (
   fieldNode: FieldNode
-): string | undefined => {
-  let field: string | undefined
-  // let fullPath: string[] = []
+): string[] => {
+  const path: string[] = []
   visit(fieldNode, {
-    // SelectionSet(node, _key, _parent, _path, _ancestors) {
-    //   node.selections.map(selection => {
-    //     if (selection.kind === "Field") {
-    //       if (selection.directives && selection.directives.length) {
-    //         if (selection.directives[0].name.value === "canonicalResource") {
-    //           fullPath.push(selection.name.value)
-    //         }
-    //       }
-    //     }
-    //   })
-
-    //   return
-    // },
-    Directive(node, _key, _parent, _path, ancestors) {
+    Field: {
+      enter(node) {
+        path.push(node.name.value)
+      },
+      leave() {
+        path.pop()
+      },
+    },
+    Directive(node) {
       if (node.name.value === "canonicalResource") {
-        const containingField = ancestors.slice(-1)[0] as FieldNode
-        field = containingField.name.value
         return BREAK
       }
     },
   })
 
-  return field
+  return path
```

Comment:
One thing left to do is to either return an empty array or `null`.

Reply:
ya, i would return `null` in the event the directive isnt used.